Octokit/Models/Response/StringEnum.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using Octokit.Internal;
+
+namespace Octokit
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    [SuppressMessage("Microsoft.Naming", "CA1711:IdentifiersShouldNotHaveIncorrectSuffix")]
+    public struct StringEnum<TEnum> : IEquatable<StringEnum<TEnum>>
+        where TEnum : struct
+    {
+        private readonly string _stringValue;
+
+        private TEnum? _parsedValue;
+
+        public StringEnum(TEnum parsedValue)
+        {
+            if (!Enum.IsDefined(typeof(TEnum), parsedValue))
+            {
+                throw GetArgumentException(parsedValue.ToString());
+            }
+
+            // Use the SimpleJsonSerializer to serialize the TEnum into the correct string according to the GitHub Api strategy
+            _stringValue = new SimpleJsonSerializer().SerializeEnum(parsedValue as Enum);
```

Comment:
Yeah that does seem like a logic flaw. The idea of the cache is to obviously "speed up" the enum deserialization by not having to evaluate the Parameter property on an enum value more than once 

But that seems to be a logic issue where only the first enum member encountered would be added to the cache...  then next time (for the same enum type) the enum type would already be in the cache (but not other members) and it would actually never look at the parameter property on other members and only do a string conversion. 

Unless I'm missing something,  this line should have actually led to bugs where we ignored the parameter attribute on any enum members other than the first one encountered? How has this not led to other problems in the past! :confused:

So yes I think removing that line is fine but now I want to understand who this hasn't already caused us problems before!